src/script/script.cpp

```diff
@@ -262,6 +263,21 @@ bool CScript::IsPayToScriptHash() const
     return p2sh || p2shWithReplay;
 }
 
+bool CScript::IsPayToPublicKey() const
+{
+    // Extra-fast test for pay-to-pubkey CScripts:
+
+    // Check if this script is P2PK without OP_CHECKBLOCKATHEIGHT
+    bool p2pk = (this->size() == 35 &&
+               (*this)[0] == 0x21 &&
+               (*this)[34] == OP_CHECKSIG);
```

Comment:
Why dereference `size` ? CScript inherits from vector and `size` is public method
```suggestion
    bool p2pk = (size() == 35 &&
               (*this)[0] == 0x21 &&
               (*this)[34] == OP_CHECKSIG);
```

Reply:
Done